app/src/main/java/com/amaze/filemanager/ui/fragments/CompressedExplorerFragment.kt

```diff
@@ -0,0 +1,643 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.amaze.filemanager.ui.fragments
+
+import android.content.ComponentName
+import android.content.ContentResolver
+import android.content.Intent
+import android.content.ServiceConnection
+import android.graphics.drawable.ColorDrawable
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.os.IBinder
+import android.provider.MediaStore
+import android.util.Log
+import android.view.*
+import android.widget.TextView
+import android.widget.Toast
+import androidx.annotation.ColorInt
+import androidx.annotation.StringRes
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.preference.PreferenceManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import com.amaze.filemanager.R
+import com.amaze.filemanager.adapters.CompressedExplorerAdapter
+import com.amaze.filemanager.adapters.data.CompressedObjectParcelable
+import com.amaze.filemanager.application.AppConfig
+import com.amaze.filemanager.asynchronous.asynctasks.AsyncTaskResult
+import com.amaze.filemanager.asynchronous.asynctasks.DeleteTask
+import com.amaze.filemanager.asynchronous.services.ExtractService
+import com.amaze.filemanager.file_operations.filesystem.OpenMode
+import com.amaze.filemanager.filesystem.HybridFileParcelable
+import com.amaze.filemanager.filesystem.compressed.CompressedHelper
+import com.amaze.filemanager.filesystem.compressed.showcontents.Decompressor
+import com.amaze.filemanager.filesystem.files.FileUtils
+import com.amaze.filemanager.ui.activities.MainActivity
+import com.amaze.filemanager.ui.colors.ColorPreferenceHelper
+import com.amaze.filemanager.ui.fragments.data.CompressedExplorerFragmentViewModel
+import com.amaze.filemanager.ui.fragments.preference_fragments.PreferencesConstants
+import com.amaze.filemanager.ui.theme.AppTheme
+import com.amaze.filemanager.ui.views.DividerItemDecoration
+import com.amaze.filemanager.ui.views.FastScroller
+import com.amaze.filemanager.utils.BottomBarButtonPath
+import com.amaze.filemanager.utils.OnAsyncTaskFinished
+import com.amaze.filemanager.utils.Utils
+import com.github.junrar.exception.UnsupportedRarV5Exception
+import com.google.android.material.appbar.AppBarLayout
+import com.google.android.material.appbar.AppBarLayout.OnOffsetChangedListener
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.util.*
+
+@Suppress("TooManyFunctions")
+class CompressedExplorerFragment : Fragment(), BottomBarButtonPath {
```

Comment:
There are quite a few `!!` used here, this should be fixed.

Reply:
> There are quite a few `!!` used here, this should be fixed.

See #2796